internal/datacoord/garbage_collector_test.go

```diff
@@ -122,13 +122,11 @@ func Test_garbageCollector_scan(t *testing.T) {
 	t.Run("key is reference", func(t *testing.T) {
 		segReferManager := &SegmentReferenceManager{
 			etcdKV: etcdKV,
-			segmentsLock: map[UniqueID][]*SegmentLock{
-				2: {
-					{
-						segmentID: 2,
-						nodeID:    1,
-						locKey:    "path",
-					},
+			segmentsLock: map[UniqueID]*datapb.SegmentReferenceLock{
```

Comment:
I was thinking add a 
SegmentID -> task list inverted index to accelerate HasSegmentLock request